backend/api/v1/project_service.go

```diff
@@ -426,7 +428,119 @@ func (s *ProjectService) SetIamPolicy(ctx context.Context, request *v1pb.SetIamP
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, err.Error())
 	}
-	return convertToV1IamPolicy(ctx, s.store, iamPolicyMessage)
+
+	v1IamPolicy, err := convertToV1IamPolicy(ctx, s.store, iamPolicyMessage)
+	if err != nil {
+		return nil, err
+	}
+	v1IamPolicy.BindingDeltas = findIamPolicyDeltas(oriIamPolicy, iamPolicyMessage.Policy)
+	return v1IamPolicy, nil
+}
+
+type bindMapKey struct {
+	User string
+	Role string
+}
+
+type condMap map[exportedExpr]bool
+
+// only contains exported fields in expr.Expr.
+type exportedExpr struct {
```

Comment:
What are the fields are should not be exported?

Reply:
<img width="433" alt="image" src="https://github.com/user-attachments/assets/f57789c0-7910-4622-b05e-dbf44847a722">

Reply:
such as state, sizeCache and unknownFields. These are fields that cannot be compared in Golang.

Reply:
Those internal fields (lower case and without field tag) are not marshal in the content.

Reply:
But the struct containing these internal fileds cannot be used as keys in the map.

Reply:
protojson.Marshal to a string as the map key?

Reply:
This approach is not very common, but it does ensure code cleanliness.

Reply:
Or we can use the expression string in the condition and ignoring the other condition fields. Either way.

Reply:
let's use protojson.Marshal because we still need the unserialized objects.